docs/tutorials/configure/pet-photographer.md

```diff
@@ -0,0 +1,730 @@
+        )
+    )
+    module = Module.from_args()
+    module.add_model_from_registry(Camera.SUBTYPE, color_filter.ColorFilterCam.MODEL)
+    await module.start()
+
+if __name__ == "__main__":
+    asyncio.run(main())
+```
+
+{{% /tab %}}
+{{% tab name="Go"%}}
+<file>main.go</file> initializes and starts the colorfilter model of the camera component.
+
+  <details>
+    <summary>Click to view sample code from <file>main.go</file></summary>
+
+```go {class="line-numbers linkable-line-numbers"}
+// Package main is a module which serves the colorfilter custom module.
+package main
+
+import (
+  "context"
+
+  "github.com/edaniels/golog"
+  "go.viam.com/utils"
+
+  "github.com/viam-labs/modular-filter-examples/colorfilter"
+  "go.viam.com/rdk/components/camera"
+  "go.viam.com/rdk/module"
+)
+
+func main() {
+  utils.ContextualMain(mainWithArgs, module.NewLoggerFromArgs("colorfilter_module"))
+}
+
+func mainWithArgs(ctx context.Context, args []string, logger golog.Logger) (err error) {
+  myMod, err := module.NewModuleFromArgs(ctx, logger)
+  if err != nil {
+    return err
+  }
+
+  err = myMod.AddModelFromRegistry(ctx, camera.API, colorfilter.Model)
+  if err != nil {
+    return err
+  }
+
+  err = myMod.Start(ctx)
+  defer myMod.Close(ctx)
+  if err != nil {
+    return err
+  }
+  <-ctx.Done()
+  return nil
+}
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+For more information on creating your own module, read [Code your own modules to create modular resources](https://docs.viam.com/modular-resources/create/).
+
+## Code your filter resource model
+
+The code for this colorfilter camera model (<file>[color_filter.go](https://github.com/viam-labs/modular-filter-examples/blob/main/colorfilter/color_filter.go)</file>) or (<file>[color_filter.py](https://github.com/viam-labs/modular-filter-examples/blob/main/pycolorfilter/color_filter.py)</file>) is sourced from the full modular filter examples available on the [Viam GitHub](https://github.com/viam-labs/modular-filter-examples/tree/main).
+
+### Include subtype's required methods
+
+In order to code a new filter resource model, you must implement the required methods outlined in the <file>[client.go](https://github.com/viamrobotics/rdk/components/camera/client.go)</file> file in the corresponding resource's directory.
+In this case, the `rdk/components/camera` directory.
+
+Provide this as a file inside of your `colorfilter` module directory to serve as your module's client interface, <file>color_filter.go</file> or <file>color_filter.py</file>. Make sure you name your model with all lowercase letters for optimal performance with Viam's SDKs.
+
+For more information, read [Create a custom module](https://docs.viam.com/modular-resources/create/#create-a-custom-module).
+
+### Include filter module requirements
+
+When creating your own filter module, it's required to check whether the data management service is the caller of the filtering to prevent unwanted effects on the filter state.
```

Comment:
This sounds great! In the `color_filter.go` file, I see that `data.FromDMContextKey` is being used in place of `data.FromDMString`. I do see `data.FromDMString` being used in the sensor filter, but not in the color filter. Since this is something you'd like to be included, I'd love to ask more about this when you're feeling better!

Reply:
@agreenb Regarding the [`Next` ](https://github.com/viam-labs/modular-filter-examples/blob/cd2f79c52b98c3deafbd1e9794869744803d4428/colorfilter/color_filter.go#L138) function in `color_filter.go`. Using `extra.FromDMString` has been emphasized, but I only see it being used in the [sensor filter](https://github.com/viam-labs/modular-filter-examples/blob/cd2f79c52b98c3deafbd1e9794869744803d4428/sensorfilter/sensor_filter.go#L98C1-L98C1). I'm very curious about the difference in their use!

Reply:
Camera is a one-off case that uses context in Go, since we don't currently have the `extra` field in our Go camera APIs — this is actually a ticket that the AV team will tackle in the next eng cycle.

So in Go, that looks like checking `ctx.Value(data.FromDMContextKey{})` as seen [here](https://github.com/viam-labs/modular-filter-examples/blob/cd2f79c52b98c3deafbd1e9794869744803d4428/colorfilter/color_filter.go#L139). We grab that from the `extra` field that can be passed in from Python, so Python will still use `from_dm_from_extra`